sopel/coretasks.py

```diff
@@ -401,6 +402,34 @@ def _send_who(bot, channel):
         # We might be on an old network, but we still care about keeping our
         # user list updated
         bot.write(['WHO', channel])
+    channel.last_who = datetime.datetime.utcnow()
+
+
+@sopel.module.interval(30)
+def _periodic_send_who(bot):
+    """Periodically send a WHO request to keep user information up-to-date."""
+    if 'away-notify' in bot.enabled_capabilities:
+        # WHO not needed to update 'away' status
+        return
+
+    # Loops through the channels to find the one that has the longest time since the last WHO
+    # request, and issues a WHO request only if the last request for the channel was more than
+    # 120 seconds ago.
+    who_trigger_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=120)
+    selected_channel = None
+    for channel in bot.channels:
+        if channel.last_who is None:
+            # WHO was never sent yet to this channel: stop here
+            selected_channel = channel
+            break
+        if channel.last_who < who_trigger_time:
+            # this channel's last who request is the most outdated one at the moment
+            selected_channel = channel
+            who_trigger_time = channel.last_who
+
+    if (selected_channel is not None):
```

Comment:
```suggestion
    if selected_channel is not None:
```

Reply:
Darn.  I missed that when I was doing the last clean-up.  Done now.